packages/react-components/react-motions-preview/src/types.ts

```diff
@@ -1,13 +1,19 @@
 export type AtomMotion = { keyframes: Keyframe[] } & KeyframeEffectOptions;
 
 export type PresenceMotion = {
-  enter: AtomMotion;
-  exit: AtomMotion;
+  enter: AtomMotion | AtomMotion[];
+  exit: AtomMotion | AtomMotion[];
 };
 
-export type AtomMotionFn = (element: HTMLElement) => AtomMotion;
+export type AtomMotionFn = (element: HTMLElement) => AtomMotion | AtomMotion[];
```

Comment:
just an idea for future: it would be nice if this function also `isReducedMotion`

Reply:
I also would like to see reduced motion be parameterized throughout. It could be an optional that defaults to the browser's setting, but overridable for testing. This would enable use cases like displaying both the regular and reduced motion versions of a motion component side-by-side.

Reply:
I would say that it depends if we want to make `isReducedMotion` declarative or not. The original proposal was to have:

```ts
const atom = {
  keyframes: [],
  duration: 500,
  reducedMotion: {
    keyframes: [],
    duration: 100,
  }
}
```

Anyway this could be added later 😉 

> It could be an optional that defaults to the browser's setting, but overridable for testing. This would enable use cases like displaying both the regular and reduced motion versions of a motion component side-by-side.

We can implement this as a React context how it was done in v0 & in some other libraries. Anyway, that's something that we can address separately 🐱